nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-ui/src/main/webapp/js/nf/canvas/controllers/nf-ng-canvas-flow-status-controller.js

```diff
@@ -400,6 +406,554 @@
+
+                                // setup violation menu handling
+                                flowAnalysisCtrl.setViolationMenuHandling(response, groupId);
+                            }
+                        }).fail(nfErrorHandler.handleAjaxError);
+                    }
+                },
+
+                /**
+                 * Set event bindings for rule menus
+                 */
+                setRuleMenuHandling: function() {
+                    $('.rule-menu-btn').click(function(event) {
+                        // stop event from immediately bubbling up to document and triggering closeRuleWindow
+                        event.stopPropagation();
+                        var ruleInfo = $(this).data('ruleInfo');
+                        $('#violation-menu').hide();
+                        $('#rule-menu').show();
+                        $('#rule-menu').position({
+                            my: "left top",
+                            at: "left top",
+                            of: event
+                        });
+
+                        // rule menu bindings
+                        $('#rule-menu-more-info').on( "click", openRuleMoreInfoDialog);
+                        $('#rule-menu-edit-rule').on('click', openRuleDetailsDialog);
+                        $(document).on('click', closeRuleWindow);
+
+                        function closeRuleWindow(e) {
+                            if ($(e.target).parents("#rule-menu").length === 0) {
+                                $("#rule-menu").hide();
+                                unbindRuleMenuHandling();
+                            }
+                        }
+
+                        function openRuleDetailsDialog() {
+                            $('#rule-menu').hide();
+                            nfSettings.showSettings().done(function() {
+                                nfSettings.selectFlowAnalysisRule(ruleInfo.id);
+                            });
+                            unbindRuleMenuHandling()
+                        }
+
+                        function openRuleMoreInfoDialog() {
+                            $('#rule-menu').hide();
+                            $('#rule-type-pill').empty()
+                                                .removeClass()
+                                                .addClass(ruleInfo.enforcementPolicy.toLowerCase() + ' rule-type-pill')
+                                                .append(ruleInfo.enforcementPolicy);
+                            $('#rule-display-name').empty().append(ruleInfo.name);
+                            $('#rule-description').empty().append(ruleInfo.descriptors['component-type'].description);
+                            $( "#rule-menu-more-info-dialog" ).modal( "show" );
+                            $('.rule-docs-link').click(function () {
+                                // open the documentation for this flow analysis rule
```

Comment:
This "View documentation" is has some error in it.
2 symptoms (I assume both require multiple enable rules of different types):
1. Many time the documentation of a wrong rule is shown.
2. When debugging this function runs multiple times.

Reply:
I'm still not sure what caused this issue, but I can't reproduce it at all. Perhaps the changes from my last commit, which moves the documentation link to the rule options menu, resolved this issue. Could you test again with the latest when you have a chance, @tpalfy?

Reply:
Please disregard my last comment. I was finally able to reproduce the issue (happens when clicking from one rule options menu trigger then immediately clicking on another with no breakpoint set).

Reply:
Thanks for pointing this out, @tpalfy. Seems like the issue was an event handler that wasn't removed properly. I've since pushed a fix for the issue.